### DISCUSSION TITLE
Test the dev environment

### DISCUSSION THREAD

**Discussion by SolitudePy** [k1nd0ne/VolWeb#26] [General]
Hello, I am trying to test VolWeb in a dev environment, I managed to installed it after installing psycopg2-binary instead, I set it up on a virtual machine and I'm trying to access it from my PC, so I set up an SSH tunnel that will let me access localhost as the virtual machine IP. I am getting weird errors when trying to create stuff such as cases,evidences such as:
```
raise ValueError("No route found for path %r." % path)
ValueError: No route found for path 'evidences/'
```
I would like to hear suggestions as to how can I bind the address more logically to 0.0.0.0 instead(tried editing the environment variables with no success), thanks for help

**Comment by SolitudePy** [comment]
I managed to bind to a different address using python manage.py 0.0.0.0:8000 but getting the same error everytime when trying to upload an evidence/ create a case. "No route found for path"

**Reply by k1nd0ne** [reply]
Hello, 
You have to configure the environment variables according to your setup. 
For example, if you are connecting to http://192.168.1.27:8000 (where 192.168.1.27 is the IP of your VM), your environment variables in your .env and  python3 virtual env should look like this: 

docker/.env
```

AWS_ENDPOINT_URL=http://192.168.1.27:9000
AWS_ENDPOINT_HOST=192.168.1.27:9000
AWS_REGION=""

AWS_ACCESS_KEY_ID=user
AWS_SECRET_ACCESS_KEY=password

BROKER_HOST=localhost
BROKER_PORT=6379
```

venv/bin/activate
```
export DEBUG=True
export CSRF_TRUSTED_ORIGINS=http://192.168.1.27:8000
export WEBSOCKET_URL=ws://192.168.1.27:8000

export AWS_ENDPOINT_URL=http://192.168.1.27:9000
export AWS_ENDPOINT_HOST=192.168.1.27:9000
export AWS_REGION=""

export AWS_ACCESS_KEY_ID=user
export AWS_SECRET_ACCESS_KEY=password

export BROKER_HOST=192.168.1.27
export BROKER_PORT=6379
```

Best regards.

**Comment by SolitudePy** [comment]
I changed all that, restarted the containers and still when I run python3 manage.py runserver 8000 I get the following:
```
INFO:windows_engine.models:Volatility3 Plugins are loaded without failure
INFO:linux_engine.models:Volatility3 Plugins are loaded without failure
INFO:windows_engine.models:Volatility3 Plugins are loaded without failure
INFO:linux_engine.models:Volatility3 Plugins are loaded without failure
Watching for file changes with StatReloader
INFO:django.utils.autoreload:Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
October 16, 2024 - 08:09:32
Django version 4.2.11, using settings 'VolWeb.settings'
Starting ASGI/Daphne version 4.1.0 development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
INFO:daphne.server:HTTP/2 support enabled
INFO:daphne.server:Configuring endpoint tcp:port=8000:interface=127.0.0.1
INFO:daphne.server:Listening on TCP address 127.0.0.1:8000
```

the minio interface is accessible through port 9000, this is the output of netstat on my vm:

```
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.0.1:8000          0.0.0.0:*               LISTEN      72906/python3
tcp        0      0 127.0.0.1:9000          0.0.0.0:*               LISTEN      72697/minio
tcp        0      0 0.0.0.0:6379            0.0.0.0:*               LISTEN      72592/docker-proxy
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      837/sshd
tcp        0      0 0.0.0.0:5432            0.0.0.0:*               LISTEN      72604/docker-proxy
tcp6       0      0 :::9090                 :::*                    LISTEN      1/systemd
tcp6       0      0 :::9000                 :::*                    LISTEN      72697/minio
tcp6       0      0 ::1:9000                :::*                    LISTEN      72697/minio
tcp6       0      0 :::9001                 :::*                    LISTEN      72697/minio
tcp6       0      0 :::6379                 :::*                    LISTEN      72598/docker-proxy
tcp6       0      0 :::22                   :::*                    LISTEN      837/sshd
tcp6       0      0 :::5432                 :::*                    LISTEN      72608/docker-proxy
udp        0      0 127.0.0.1:323           0.0.0.0:*                           821/chronyd
udp6       0      0 ::1:323                 :::*                                821/chronyd
```

**Reply by k1nd0ne** [reply]
You are listening on 127.0.0.1 (localhost). 
So if you are accessing the VolWeb interface on http://localhost:8000 the env variable shoud be updated with localhost:8000 and localhost:9000. 

If you are accessing the interface from an other machine you need to make django listen on 0.0.0.0:8000 and update your env accordingly like mentioned before.

Regards.

**Comment by SolitudePy** [comment]
it is setup on a linux vm so I am trying to access it from my PC, as I said before I tried running it with python3 manage.py runserver <vm-ip>:8000 and I managed to access it from my PC but getting routing errors as mentioned before. is there other way to make django listen without it breaking your app?
